File: python_code/IntersectionOfTwo.py
class IntersectionOfTwo:
    @staticmethod
    def intersect(nums1, nums2):
        from collections import Counter

        freq = Counter(nums1)  # ② contar ocurrencias de nums1
        out = []  # ④ lista para almacenar la intersección

        # ⑤ recorrer nums2 y verificar intersección
        for num in nums2:
            if freq.get(num, 0) > 0:  # ⑥
                out.append(num)  # ⑦
                freq[num] -= 1  # ⑧

        return out  # ⑪

# Ejemplo de uso
nums1 = [1, 2, 2, 1]
nums2 = [2, 2]
result = IntersectionOfTwo.intersect(nums1, nums2)
print("Intersección:", result)  # Resultado esperado: [2, 2]

File: ts_code/IntersectionOfTwo.ts
class IntersectionOfTwo {
    public static intersect(nums1: number[], nums2: number[]): number[] {
        const freq: Map<number, number> = new Map(); // ②

        // ③
        for (const num of nums1) {
            freq.set(num, (freq.get(num) || 0) + 1);
        }

        const out: number[] = []; // ④

        // ⑤
        for (const num of nums2) {
            if ((freq.get(num) || 0) > 0) { // ⑥
                out.push(num); // ⑦
                freq.set(num, freq.get(num)! - 1); // ⑧
            }
        }

        return out; // ⑪
    }
}

// Ejemplo de uso:
const nums1 = [1, 2, 2, 1];
const nums2 = [2, 2];
const results = IntersectionOfTwo.intersect(nums1, nums2);
console.log("Intersección:", results); // Resultado esperado: [2, 2]
